torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -554,19 +552,36 @@ def _compute_affine_output_size(matrix: List[float], w: int, h: int) -> Tuple[in
 
 
 def _apply_grid_transform(img: torch.Tensor, grid: torch.Tensor, mode: str, fill: _FillTypeJIT) -> torch.Tensor:
+    input_shape = img.shape
```

Comment:
Putting the handling avoids repeating this in every kernel that uses this as core action, i.e. `F.affine`, `F.rotate`, `F.perspective`, and `F.elastic`.